aspnetcore/fundamentals/portable-object-localization.md

```diff
@@ -0,0 +1,218 @@
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/localization/sample/POLocalization)
+
+## What is a PO file?
+
+PO files contain the translated strings for a given language. They prove very useful, as opposed to standard *.resx* files. PO files support pluralization and are distributed as plain text files.
+
+### Example
+
+Here is a sample PO file containing the translation for two strings in French, including one with its plural form:
+
+*fr.po*
+
+```text
+#: Services/EmailService.cs:29
+msgid "Enter a comma separated list of email addresses."
+msgstr "Entrez une liste d'emails séparés par une virgule."
+
+#: Views/Email.cshtml:112
+msgid "The email address is \"{0}\"."
+msgid_plural "The email addresses are \"{0}\"."
+msgstr[0] "L'adresse email est \"{0}\"."
+msgstr[1] "Les adresses email sont \"{0}\""
+```
+
+This example uses the following syntax:
+
+- `#:`: A comment indicating the context of the string to be translated. The same string might be translated differently depending on where it is being used.
+- `msgid`: The untranslated string.
+- `msgstr`: The translated string.
+
+In the case of pluralization support, more entries can be defined.
+
+- `msgid_plural`: The untranslated plural string.
+- `msgstr[0]`: The translated string for the case 0.
+- `msgstr[N]`: The translated string for the case N.
+
+The PO file specification can be found [here](https://www.gnu.org/savannah-checkouts/gnu/gettext/manual/html_node/PO-Files.html).
+
+## Configuring PO file support in ASP.NET Core
+
+This example is based on an ASP.NET Core MVC application generated from a Visual Studio 2017 project template.
+
+### Referencing the package
+
+Add a reference to the `OrchardCore.Localization.Core` NuGet package. It is available on [MyGet](https://www.myget.org/) at the following package source: https://www.myget.org/F/orchardcore-preview/api/v3/index.json
+
+The *.csproj* file now contains a line similar to the following (version number may vary):
+
+[!code-xml[Main](localization/sample/POLocalization/POLocalization.csproj?range=9)]
+
+### Registering the service
+
+Add the required services to the `ConfigureServices` method of *Startup.cs*:
+
+[!code-csharp[Main](localization/sample/POLocalization/Startup.cs?name=snippet_ConfigureServices&highlight=4-21)]
+
+Add the required middleware to the `Configure` method of *Startup.cs*:
+
+[!code-csharp[Main](localization/sample/POLocalization/Startup.cs?name=snippet_Configure&highlight=15)]
+
+Add the following code to your Razor view of choice. *About.cshtml* is used in this example.
+
+[!code-cshtml[Main](localization/sample/POLocalization/Views/Home/About.cshtml)]
+
+An `IViewLocalizer` instance is injected and used to translate the text `"Hello world!"`.
+
+### Creating a PO file
+
+Create a file named *fr.po*, in your application root folder, containing the following:
```

Comment:
Can we make this generic?
Create a file with the name `<Country Code>.po` (or whatever fr is in this case). In this example, French is used so the file is named *fr.po*.